### ISSUE TITLE
Incorrect local variable type inference

### ISSUE THREAD

**Issue by cushon** [typetools/checker-framework#475]
The compilation succeeds if `xs` is annotated as `@Initialized @NonNull`, so it looks like the type of the local variable is being inferred incorrectly.

``` java
import java.util.Set;
import com.google.common.collect.ImmutableSet;

class Test {
  private static final Set<String> A = ImmutableSet.of();
  private static final Set<String> B = ImmutableSet.of(init(""));

  public static String init(String s) { return s; }

  public static void g(Set<String> xs) {}

  void m(boolean flag) {
    Set<String> xs = flag ? A : B;
    g(xs);
  }
}
```

```
$ ~/jsr308/checker-framework-1.9.5/checker/bin/javac -fullversion
javac full version "1.8.0-jsr308-1.9.5-b00"
$ wget http://central.maven.org/maven2/com/google/guava/guava/18.0/guava-18.0.jar
$ ~/jsr308/checker-framework-1.9.5/checker/bin/javac -cp guava-18.0.jar -processor org.checkerframework.checker.nullness.NullnessChecker -AprintErrorStack -AprintAllQualifiers Test.java
Test.java:14: error: [argument.type.incompatible] incompatible types in argument.
    g(xs);
      ^
  found   : @UnknownInitialization @Nullable Set<@Initialized @NonNull String>
  required: @Initialized @NonNull Set<@Initialized @NonNull String>
1 error
```

**Comment by smillst** [comment]
I can no longer reproduce this.  I believe the fixes for #643 (LUB) likely fixed this bug. (The type of the conditional expression `flag ? A : B` was probably computed incorrectly before this fix.
